src/Oauth1.php

```diff
@@ -96,10 +104,10 @@ public function onBefore(BeforeEvent $event)
         switch ($this->config['request_method']) {
             case self::REQUEST_METHOD_HEADER:
                 list($header, $value) = $this->buildAuthorizationHeader($params);
-                $request->setHeader($header, $value);
+                $request = $request->withHeader($header, $value);
                 break;
             case self::REQUEST_METHOD_QUERY:
-                $request->getQuery()->overwriteWith($params);
+                $request = $request->withUri($request->getUri()->withQuery(\GuzzleHttp\Psr7\build_query($params)));
```

Comment:
This previously would merge params on top of the existing query string parameters. Now it looks like this does a full replace of all existing query string parameters. Is that intentional?